static/app/views/insights/mobile/screens/components/screensOverview.tsx

```diff
@@ -156,63 +161,57 @@ export function ScreensOverview() {
     }
   }, [primaryData]);
 
-  const tableSearchFilters = new MutableSearch(['transaction.op:ui.load']);
-  const derivedQuery = getTransactionSearchQuery(location, primaryEventView.query);
-
-  const combinedData = useMemo((): TableData | undefined => {
-    if (defined(primaryData) && defined(secondaryData)) {
-      const meta: MetaType = {};
-      meta.units = {
-        ...secondaryData.meta?.units,
-        ...primaryData.meta?.units,
-      };
-      meta.fields = {
-        ...secondaryData.meta?.fields,
-        ...primaryData.meta?.fields,
-      };
-
-      const data = primaryData.data.map(row => {
-        const matchingRow = secondaryData.data.find(
-          metricRow => metricRow.transaction === row.transaction
-        );
-        if (matchingRow) {
-          return {
-            ...matchingRow,
-            ...row,
-          };
-        }
-        return row;
-      });
-
-      return {
-        data,
-        meta,
-      };
-    }
-
-    return primaryData;
-  }, [primaryData, secondaryData]);
+  const derivedQuery = getTransactionSearchQuery(location, primaryQuery);
+
+  const combinedData = useMemo(() => {
+    const meta: MetaType = {};
+    meta.units = {
+      ...secondaryMeta?.units,
+      ...primaryMeta?.units,
+    };
+    meta.fields = {
+      ...secondaryMeta?.fields,
+      ...primaryMeta?.fields,
+    };
+
+    const data = primaryData.map(row => {
+      const matchingRow = secondaryData.find(
+        metricRow => metricRow.transaction === row.transaction
+      );
+      if (matchingRow) {
+        return {
+          ...matchingRow,
+          ...row,
+        };
+      }
+      return row;
+    });
+
+    return {
+      data,
+      meta,
+    };
+  }, [primaryData, secondaryData, primaryMeta, secondaryMeta]);
 
   const loading = primaryLoading || (hasVisibleScreens && secondaryLoading);
   return (
     <Container>
-      <SearchBar
-        eventView={primaryEventView}
-        onSearch={search => {
-          router.push({
-            pathname: router.location.pathname,
-            query: {
-              ...location.query,
-              cursor: undefined,
-              query: String(search).trim() || undefined,
-            },
-          });
-        }}
-        organization={organization}
-        query={getFreeTextFromQuery(derivedQuery)!}
-        placeholder={t('Search for Screen')}
-        additionalConditions={tableSearchFilters}
-      />
+      {!loading && (
```

Comment:
🤔 For some odd reason, if this `loading` check isn't here I get 
![image](https://github.com/user-attachments/assets/fe678078-c2ac-4364-99b6-020a53992ed8)